src/waveform/renderers/waveformwidgetrenderer.cpp

```diff
@@ -288,3 +288,19 @@ void WaveformWidgetRenderer::setTrack(TrackPointer track) {
         m_rendererStack[i]->onSetTrack();
     }
 }
+
+WaveformMarkPointer WaveformWidgetRenderer::getCueMarkAtPoint(QPoint point) {
+    for (const auto& pMark : m_markPositions.keys()) {
+        int markImagePositionInWidgetSpace = m_markPositions[pMark];
+        QPoint pointInImageSpace;
+        if (getOrientation() == Qt::Horizontal) {
+            pointInImageSpace = QPoint(point.x() - markImagePositionInWidgetSpace, point.y());
+        } else { /* Vertical */
```

Comment:
You could replace this comment with `DEBUG_ASSERT(getOrientation() == Qt::Vertical);` to document **and** verify your assumptions.